### ISSUE TITLE
module naming

### ISSUE THREAD

**Issue by ckot** [peterpeter5/pyredux#1]
Hi,

First off, awesome work!  I was just searching around to see if someone has implemented something like this prior to implementing my own.

While it doesn't prevent things from working, module names are supposed to be lower case.  I understand that this would be a breaking change, but was wondering if you could fix this (it might cause problems on some platforms).  Unfortunately, I don't know if there is any way to tell how many current users of your package are out there either via github or PyPi.

*Perhaps* creating lower-case named versions of your modules, moving the code into them, and have the upper-cased ones simply import the stuff from the lower-cased ones, and deprecate the upper-cased modules would be a way to fix this without breaking anything.

Just a thought.

-Scott

**Comment by peterpeter5** [comment]
Hi Scott,

Thank you for bringing this up! Pep8 is realy clear at this point( which I didnt knew) I will change it tomorrow and will make a 1.0.0 release.

**Comment by ckot** [comment]
Yeah, the PEPs unfortunately specify a lot of things, some of which I merely configure my linter to ignore, but module naming is definitely one thing I adhere to.   It seems that there are some other package and or module/naming conventions I think '.' in package names or something (not applicable to your project) which broke pip installs of some of the zope ecosystem a while back, so unlike variable names, etc. I figured it would be good to let you know about that.

**Comment by peterpeter5** [comment]
Hi Scott, 

i fixed it, but I don't agree with pep8 on this topic. It does not look nicer or cleaner to me. The second point is, that by naming modules lowercase (with possible underscores) the similarity between class-methods and functions defined in a module is obfusacted.

Before you could use both in the same way, if you import the right. But:

```py
from pyredux import actions

actions.create_typed_actioncreator()
``` 
is know a breaking api-change when you want to  refactor all functions into a class with classmethods.

The third point: when you run flake8 on the version with uppercase-modules, no error or warning is displayed...

Don't get me wrong. I know (and accept) that packages have to be lowercase. If you don't do this, you can not distribute your package with conda and nearly every newer python-package is written that way.

However I  looked at the python3 - standard  - library, django and openstack and they all follow the lowercase-module convention so I decided to change it. 

I released a a new version (1.0.1) including a python3 - version with a reduced dependency-list

**Comment by ckot** [comment]
Hi Peter,  

 

I here ya.  I'm not as keen on the PEPs which constrain the programmer as the ones which are requests for adding new features to the language. I just thought I should let you know about this.  Thanks for fixing it.

 

BTW, as I mentioned earlier, I found your project when doing a search to make sure I wasn't re-inventing the wheel.   I have a few ideas I'd like to bounce off you, and perhaps become a contributor.  Do you know any way for us to share contact info without it being exposed publicly?

 

Oh, and I just noticed that you recently started creating packages for your pyredux-* ecosystem.  Cool!

 

 -Scott
